src/libs/sfdk/dockercontainer.cpp

```diff
@@ -0,0 +1,361 @@
+        arguments.append(VOLUME);
+        arguments.append(paths.first + ":" + paths.second);
+    }
+
+    arguments.append(q->name());
+
+    auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+    QObject::connect(runner.get(), &ProcessRunner::done, q, [q, functor, process = runner->process()](bool ok) {
+        if (ok)
+            q->m_containerId = QString::fromLocal8Bit(process->readAllStandardOutput().trimmed());
+        functor(ok);
+    });
+    SdkPrivate::commandQueue()->enqueue(std::move(runner));
+}
+
+void DockerContainerPrivate::stop(const QObject *context, const Functor<bool> &functor)
+{
+    Q_Q(DockerContainer);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    const QPointer<const QObject> context_{context};
+
+    auto enqueue = [=](const QStringList &arguments, CommandQueue::BatchId batch) {
+           auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+           QObject::connect(runner.get(), &ProcessRunner::failure, Sdk::instance(), [=]() {
+               SdkPrivate::commandQueue()->cancelBatch(batch);
+               callIf(context_, functor, false);
+           });
+           ProcessRunner *runner_ = runner.get();
+           SdkPrivate::commandQueue()->enqueue(std::move(runner));
+           return runner_;
+       };
+
+    const CommandQueue::BatchId batch = SdkPrivate::commandQueue()->beginBatch();
+
+    QStringList stopArguments;
+    stopArguments.append(STOP);
+    stopArguments.append(q->m_containerId);
+
+    enqueue(stopArguments, batch);
+
+    QStringList commitArguments;
+    commitArguments.append(COMMIT);
+    commitArguments.append(q->m_containerId);
+    commitArguments.append(q->name());
+
+    enqueue(commitArguments, batch);
+
+    SdkPrivate::commandQueue()->enqueueCheckPoint(context, [=]() { functor(true); });
+    SdkPrivate::commandQueue()->endBatch();
+}
+
+void DockerContainerPrivate::probe(const QObject *context,
+        const Functor<VirtualMachinePrivate::BasicState, bool> &functor) const
+{
+    Q_Q(const DockerContainer);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    DockerManager::fetchRegisteredVirtualMachines(q, [=](const QStringList &registeredVms,
+                                                         bool ok) {
+        VirtualMachinePrivate::BasicState state;
+        if (!ok) {
+            functor(state, false);
+        } else if (!registeredVms.contains(q->name())) {
+            functor(state, true);
+        } else {
+            QStringList arguments;
+            arguments.append(CONTAINER_LIST);
+            arguments.append(QString(FILTER_PREFIX) + q->name());
+            arguments.append(FORMAT);
+            auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+            QObject::connect(runner->process(),
+                QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished), context,
```

Comment:
Should be indented once more

Reply:
No, actually it was the next line which had wrong indentation level. Please check if you like the current state.

Reply:
Argument list (first level parentheses) should be indented twice, so really, this line was wrong. If this is the current code https://github.com/sailfishos/sailfish-qtcreator/pull/223/files#diff-8c6731e3f6adcdd8ada5e49b9f656c07R229, then there is a newer comment with suggested wrapping.

Reply:
Fixed now according to the other comment